### PR TITLE
Initial CMake Support

### PR DESCRIPTION
**Pull request by brianretford**
Requires the following external components to be installed presently:

* cmake (a modern version)
* Probably boost (TODO: use a cmake Fetch for this)
* python
  * pystache
* ???

TODO: get all the external deps to be managed within cmake, and failing that, an external conda environment.

```
mkdir build-opt
cd build-opt
cmake .. -G Ninja -D CMAKE_BUILD_TYPE=Release
ninja && ninja check-pml && bin/cc_test
```

### CODE REVIEW

File: pmlc/target/x86/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+pml_mlir_transforms(
+  SRCS
+    "heatmap_impl.cc"
```
Comment (jbruestle) [inline pmlc/target/x86/CMakeLists.txt:3]: Why the quotes?

Comment (flaub) [reply pmlc/target/x86/CMakeLists.txt:3]: This was an IREE convention, but as I'm refactoring, I'll remove the quotes (they don't seem to be necessary)

File: pmlc/dialect/pxa/CMakeLists.txt [hunk 2]
```diff
@@ -1,14 +1,4 @@
...
-pml_cc_library(
-  NAME
-    pxa 
-  DEPS
-    ::analysis
-    ::ir
-    ::transforms
-)
\ No newline at end of file
+add_subdirectory(transforms)
```
Comment (flaub) [inline pmlc/dialect/pxa/CMakeLists.txt:4]: Please add newlines at end of file.

File: pmlc/dialect/tile/ir/CMakeLists.txt [hunk 3]
```diff
@@ -15,6 +15,7 @@ pml_cc_library(
     ops.cc
     util.cc
   DEPS
+    pmlc::dialect::stdx::transforms
```
Comment (flaub) [inline pmlc/dialect/tile/ir/CMakeLists.txt:18]: Tile should not depend on STDX

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  cmake_policy(SET CMP0075 NEW)
+endif()
+
+if(POLICY CMP0077)
+  cmake_policy(SET CMP0077 NEW)
+endif()
+
+include(FetchContent)
+
+# set the project name
```
Comment (brianretford) [inline CMakeLists.txt:18]: remove extraneous

**Review by brianretford — COMMENTED** [review]
Looks great.

**Review by flaub — APPROVED** [review]

**Pull request merged**